### PR TITLE
add support to config SuccessfulJobsHistoryLimit (#618)

### LINKED ISSUE #618: Add SuccessfulJobsHistoryLimit config support for cronjobs
Hi All,

I add SuccessfulJobsHistoryLimit config for cronjobs, so that we can control the number of successful cron jobs history. 

I have tested it in my test env. Please help me to review and merge it. Thanks

### PR DESCRIPTION
**Pull request by peacecoder**
SuccessfulJobsHistoryLimit will control the number of cronjobs history after execute successfully.
This is very useful when some monitor service like prometheus will be set the alarm if the job status is not allow to be completed.

Closes #618

### CODE REVIEW

**Comment by peacecoder** [comment]
> Thanks for your contribution! It looks great, just need some documentation. It would also be nice to add this new property to a test, even if just to an existing one.
> 
> About documenting this new property: could you please either change an existing example to include this option or create a PR against the docs repository, documenting this new property?
> 
> https://github.com/jaegertracing/documentation/ , and the content file is this one: `content/docs/next-release/operator.md`

Thanks for your review, I will add some it to the test and change documentation as your suggestion

**Comment by jpkrohling** [comment]
@peacecoder do you have news about the tests? Would love to get this one merged!

**Comment by peacecoder** [comment]
> @peacecoder do you have news about the tests? Would love to get this one merged!

Hi  @jpkrohling , I'am so sorry about late response. I change the doc firstly, please help me to review https://github.com/jaegertracing/documentation/pull/302

I will add some test after works.

**Comment by peacecoder** [comment]
> @peacecoder do you have news about the tests? Would love to get this one merged!

Hi @jpkrohling , 

I have add some test and documentation about this change, please help me to review it. thanks.

https://github.com/jaegertracing/documentation/pull/302

**Comment by jpkrohling** [comment]
@peacecoder could you please run `make generate` and update this PR? The CI job is failing because a model has been changed:

```
Build failed: a model has been changed but the deep copy functions aren't up to date. Run 'make generate' and update your PR.
```

**Comment by peacecoder** [comment]
Thanks, pass all ci checks.

File: pkg/cronjob/es_index_cleaner_test.go [hunk 1]
```diff
@@ -16,13 +16,17 @@ func TestCreateEsIndexCleaner(t *testing.T) {
 		map[string]interface{}{"es.index-prefix": "tenant1", "es.server-urls": "http://nowhere:666,foo"})}}}
 	days := 0
 	jaeger.Spec.Storage.EsIndexCleaner.NumberOfDays = &days
+	historyLimits := int32(1)
+	assert.Nil(t, jaeger.Spec.Storage.EsIndexCleaner.SuccessfulJobsHistoryLimit)
```
Comment (jpkrohling) [inline pkg/cronjob/es_index_cleaner_test.go]: That's not really what I meant, but it just made me realize that the `assert.Nil` that I requested before didn't make much sense either. Could you revert this one?

Comment (peacecoder) [reply pkg/cronjob/es_index_cleaner_test.go]: done

**Comment by jpkrohling** [comment]
Thank you for your contribution!

**Comment by peacecoder** [comment]
My pleasure, and thank you for helping me review the code.

**Pull request merged**